### ISSUE TITLE
Running the test suite is not idempotent

### ISSUE THREAD

**Issue by danhorst** [projecthydra-labs/hyrax#295]
Running the test suite using `bundle exec rspec` or `bundle exec rake spec` will only complete successfully the _first_ time it is run. Running the same command a second time will fail with:
```ruby
NameError: uninitialized constant Catapult
```

This is because while the test suite is running `config/initializers/hyrax.rb` gets modified to include:
```ruby
# Injected via `rails g hyrax:work Catapult`
config.register_curation_concern :catapult
```
This change is not reverted afterwards.

This problem isn’t present using `bundle exec rake ci` or in the Travis builds because the test app is rebuilt every time the tests are run.

---

I would like to be able to run one spec at a time using `bundle exec rspec TEST_FILE` while I am working on a particular feature. My current workaround is to keep the contents of `.internal_test_app` tracked in git so it is easy to revert changes to `config/initializers/hyrax.rb` when needed. I only use the `ci` task to check for regression failures because it takes about 12 minutes to complete on my development machine.

Places to look: `spec/features/work_generator_spec.rb` specifically line 18:
```ruby
Rails::Generators.invoke('hyrax:work', ['Catapult'], behavior: :revoke, destination_root: Rails.root)
```

**Comment by atz** [comment]
If we separate out a new ticket "hyrax:work generator is not reversible", we can solve the test case with a basic hack-around.  Just copy the `hyrax.rb` initializer before the test and copy back after.

**Comment by jcoyne** [comment]
Fixed by #302
